kodilla-hibernate/src/test/java/com/kodilla/hibernate/manytomany/dao/facade/ManyToManyFacadeTestSuite.java

```diff
@@ -0,0 +1,111 @@
+package com.kodilla.hibernate.manytomany.dao.facade;
+
+import com.kodilla.hibernate.manytomany.Company;
+import com.kodilla.hibernate.manytomany.Employee;
+import com.kodilla.hibernate.manytomany.facade.ManyToManyFacade;
+import com.kodilla.hibernate.manytomany.facade.ManyToManyFacadeException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ManyToManyFacadeTestSuite {
+    @Autowired
+    ManyToManyFacade manyToManyFacade;
+
+    @Before
+    public void createData() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smith");
+        Employee stephanieClarckson = new Employee("Stephanie", "Clarckson");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+
+        Company softwareMachine = new Company("Software Machine");
+        Company dataMaesters = new Company("Data Maesters");
+        Company greyMatter = new Company("Grey Matter");
+
+        softwareMachine.getEmployees().add(johnSmith);
+        dataMaesters.getEmployees().add(stephanieClarckson);
+        dataMaesters.getEmployees().add(lindaKovalsky);
+        greyMatter.getEmployees().add(johnSmith);
+        greyMatter.getEmployees().add(lindaKovalsky);
+
+        johnSmith.getCompanies().add(softwareMachine);
+        johnSmith.getCompanies().add(greyMatter);
+        stephanieClarckson.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(greyMatter);
+
+        manyToManyFacade.save(softwareMachine);
+        manyToManyFacade.save(dataMaesters);
+        manyToManyFacade.save(greyMatter);
+    }
+
+    @Test
+    public void facadeSearchCompaniesByThreeCharsOfNameTest() {
+        //When
+        List<Company> searchedCompaniesByThreeChars = new ArrayList<>();
+        try {
+            searchedCompaniesByThreeChars = manyToManyFacade.searchCompaniesByThreeCharsOfName("sof");
+        } catch (ManyToManyFacadeException e) {
+            e.printStackTrace();
+        }
```

Comment:
ten try-catch jest zbedny. Jak poleci wyjatek, to test i tak nie przejdzie

Reply:
mozesz pozbyc sie try-catch'a przez deklaracje testu
public void facadeSearchCompaniesByThreeCharsOfNameTest() throws ManyToMany... {